synapse/lib/stormlib/aha.py

```diff
@@ -49,13 +49,35 @@ class AhaLib(s_stormtypes.Lib):
          'type': {'type': 'function', '_funcname': '_methAhaList', 'args': (),
                   'returns': {'name': 'Yields', 'type': 'list',
                               'desc': 'The AHA service dictionaries.', }}},
+        {'name': 'callPeerApi', 'desc': '''Call an API on an AHA service.
+
+        Examples:
+            Call getCellInfo on an AHA service::
+
+                for $info in $lib.aha.callPeerApi(cortex..., getCellInfo) { $lib.print($info) }
+
+        ''',
+         'type': {'type': 'function', '_funcname': '_methCallPeerApi',
+                  'args': (
+                      {'name': 'svcname', 'type': 'str',
+                       'desc': 'The name of the AHA service to call. It is easiest to use the relative name of a service, ending with "...".', },
+                      {'name': 'apiname', 'type': ('str', 'tuple'),
+                       'desc': 'The name of the API to call or a todo tuple (name, args, kwargs).'},
+                      {'name': 'timeout', 'type': 'int', 'default': None,
+                       'desc': 'Optional timeout in seconds.'},
+                      {'name': 'skiprun', 'type': 'bool', 'default': None,
+                       'desc': 'Optional flag to skip run checks.'},
+                  ),
+                  'returns': {'type': 'any',
+                             'desc': 'The result of the API call.', }}},
     )
     _storm_lib_path = ('aha',)
     def getObjLocals(self):
         return {
             'del': self._methAhaDel,
             'get': self._methAhaGet,
             'list': self._methAhaList,
+            'callPeerApi': self._methCallPeerApi,
```

Comment:
Should we add `callPeerGenr` too?

Reply:
I don't use the `callPeerGenr` function in the tool and so left it out. This PR is a little tricky to review because it should be layered on the Gather PR which has additional code like this function for future things.

Reply:
Can you re-target the PR to the gather branch? Maybe that would make your changes vs visi's changes more clear?

Reply:
I'm going to raise this today and hope we can get the Gather PR over the finish line because it's getting unwieldy to deal with the merge conflicts.